src/Type/Definition/EnumType.php

```diff
@@ -183,10 +190,10 @@ public function assertValid(): void
 
         $values = $this->config['values'] ?? null;
         // @phpstan-ignore-next-line should not happen if used correctly
-        if (! is_iterable($values)) {
+        if (! is_iterable($values) && ! is_callable($values)) {
             $notIterable = Utils::printSafe($values);
 
-            throw new InvariantViolation("{$this->name} values must be an iterable, got: {$notIterable}");
+            throw new InvariantViolation("{$this->name} values must be an iterable or callable, got: {$notIterable}");
```

Comment:
Please add a test case to cover this, I think [here](https://github.com/webonyx/graphql-php/blob/b4cbc3ab27d35e18f4fd9318a855952aa8711875/tests/Type/DefinitionTest.php#L1703) would be a good spot.

Reply:
Done.